lein-cljfmt/src/leiningen/cljfmt.clj

```diff
@@ -6,17 +6,17 @@
             [leiningen.core.main :as main]
             [leiningen.cljfmt.diff :as diff]))
 
-(defn clojure-file? [file]
-  (re-find #"\.clj[sx]?" (str file)))
+(defn clojure-file? [pattern file]
+  (re-find pattern (str file)))
 
-(defn clojure-files [dir]
-  (filter clojure-file? (file-seq (io/file dir))))
+(defn clojure-files [pattern dir]
+  (filter (partial clojure-file? pattern) (file-seq (io/file dir))))
 
-(defn find-files [f]
+(defn find-files [pattern f]
```

Comment:
This is a function specific to the plugin, so rather than passing in the pattern, let's just accept the project:

```clojure
(defn file-pattern [project]
  (get-in project [:cljfmt :file-pattern] #"\.clj[sx]?"))

(defn find-files [project file]
  (let [f (io/file f))]
    (when-not (.exists f) (main/abort "No such file:" (str f)))
     (if (.isDirectory f)
       (grep (file-pattern project) f)
       [f])))
```